### DISCUSSION TITLE
Optimal way to share fetched data between pages?

### DISCUSSION THREAD

**Discussion by thanosoncode** [vercel/next.js#48850] [Help] [answered]
### Summary

Let's say we fetch some data on the server for Page 1 but we also want to use the same data in Page 2 without having to make the extra call. What would be the optimal way to do it? 

If it was just client and React, we could have 2 useQueries, one for each page,  and whichever got visited first would download the data and the other one when it got visited would use it from the cache.

What would be the equivalent of that scenario in Next?

**Comment by icyJoseph** [comment, marked as answer]
The short, hot take, answer: the same approach

The long answer: since Next.js has data-fetching built-in,

- Pre-rendered pages fetch the data at build time, and could update it at runtime
- Through GSSP it is possible to fetch on the server, and update a client side cache with the data
- In Server Components, the idea is that fetch would cache and dedupe requests for the same data, so, potentially no client fetching is ever needed

**Reply by thanosoncode** [reply]
How would you do the getQueryData part (or its equivalent) in Server Components though?

Bring in RTK and set/get to store without hooks maybe?

**Reply by icyJoseph** [reply]
That's where it gets spicy because, you essentially have bits that exist only on the server, and bits that run on both. So you'd need a client component to inject the Query Provider into the tree, and then by passing data as serialisable props from Server Components to Client components, you could set query data, or read from it. 

Also, a Server Component has no interactivity, so how would you read data on it? Just fetch what you need. React-Query is purely for client side work.

**Comment by thanosoncode** [comment]
I see. That's for the insights
